servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/TransportObserverInitializer.java

```diff
@@ -16,46 +16,93 @@
 package io.servicetalk.transport.netty.internal;
 
 import io.servicetalk.transport.api.ConnectionObserver;
+import io.servicetalk.transport.api.ConnectionObserver.SecurityHandshakeObserver;
 import io.servicetalk.transport.api.TransportObserver;
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufHolder;
 import io.netty.channel.Channel;
 import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler.Sharable;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
 import io.netty.channel.ChannelPromise;
+import io.netty.handler.ssl.SslHandshakeCompletionEvent;
 
+import javax.net.ssl.SSLSession;
+
+import static io.servicetalk.transport.netty.internal.NettyPipelineSslUtils.extractSslSession;
+import static io.servicetalk.transport.netty.internal.TransportObserverUtils.assignConnectionError;
 import static io.servicetalk.transport.netty.internal.TransportObserverUtils.assignConnectionObserver;
+import static io.servicetalk.transport.netty.internal.TransportObserverUtils.securityHandshakeObserver;
 import static java.util.Objects.requireNonNull;
 
 /**
  * A {@link ChannelInitializer} that registers a {@link ConnectionObserver} for all channels.
  */
 public final class TransportObserverInitializer implements ChannelInitializer {
 
+    /**
+     * Tells which side is using secure connection.
+     */
+    public enum SecureSide {
```

Comment:
The name is confusing since TLS can not be on one side.
I think this enum is overloaded to give two piece of information:
- Whether TLS is enabled.
- Whether this initializer is used by client/server.

To avoid confusion caused by this overload, I would suggest splitting this into two fields:
- `boolean isSecure`
- `boolean forClient`

This is similar to what we do for security config.